### ISSUE TITLE
Cmd-L to jump to the specific line

### ISSUE THREAD

**Issue by zrzka** [omz/Pythonista-Issues#376]
Would be nice if `Cmd L` shows some little dialog (or any other way) where I can type line number and editor will scroll to this line

**Comment by dgelessus** [comment]
Isn't Cmd+L the shortcut for the outline already? Might be a good thing though - the search field you suggested in #375 could also accept line numbers.

**Comment by zrzka** [comment]
Yup, it is. As Ole tries to sync shortcuts with Xcode default ones, would be nice to sync these somehow as well. If this is the real goal. Xcode uses `Ctrl 6` for outline.

**Comment by zrzka** [comment]
Re line number - I can imagine it can work with the outline as well. An example:

* Search field content: `123`
* Outline table content: `Jump to line 123` as a first thing, then remaining items (if they do contain 123)

**Comment by lukaskollmer** [comment]
I don't know if @omz collects any statistics on this but you also have to consider that changing the current behaviour of `Cmd+L` might break a lot of people's workflow.

I'd propose:
1) re-mapping `Cmd+L` to "Jump to line", and thus creating parity w/ Xcode
2) mapping `Cmd+Shift+L` to "View outline"

**Comment by zrzka** [comment]
@lukaskollmer in what way it will change the current behavior of `Cmd L` if there will be search field (with line numbers support)?

**General**

* Search field as a first responder
* Initially, search field is empty -> show all outline items as Pythonista does now
* Still allow arrow keys to change selection
* Enter still scrolls to the selected outline item

There's no change, it just adds ability to filter items.

**Number in the search field**

Same behavior, but when Pythonista detects number (let's say 123) in the search field, it can:

* Filter outline items in the same way (show just items containing 123)
* Prepend this list with one special item named `Jump to line 123` and if I hit Enter, it will just scroll to this line

No shortcut change, no behavior change, just enhancements.

**Comment by omz** [comment]
This isn't really a priority right now, but here's how I imagine things to work eventually:

* Cmd+L and Ctrl+6 could do the same thing (for consistency with Xcode)
* If the editor is active when the shortcut is used *or* a hardware keyboard is connected, a search field in the outline would become active (first responder) automatically, filtering the outline (using fuzzy matching).
* If a number is entered, that could be interpreted as a line number (though I don't navigate to specific line numbers very often, so I don't think that part is very important).
* If the outline is invoked while the on-screen keyboard is inactive (and no hardware keyboard is connected), the search/filter field wouldn't become active automatically (it could be activated manually though).

**Comment by zrzka** [comment]
Nice way with the first responder & HW keyboard. I didn't realize that half of the screen will be hidden by the SW keyboard.
